README.md

```diff
@@ -1,2 +1,178 @@
+  string account_id = 1;
+}
+
+message UserProfileResponse {
+  string email = 1;
+  string first_name = 2;
+  string last_name = 3;
+  classtime.service.common.Role role = 4;
+}
+```
+
+## URLs construction
+
+Any desired endpoint constructs of 3 joined blocks:
+- Our server public host: https://www.classtime.com/service/public/
+- Service name (e.g. Account, School, Library)
+- Service method name (e.g. createAccount, login, associateTeacher)
+
+Based on [interface sample](https://github.com/gopollock/public-api/tree/main#interface-sample), we could specify such endpoints:
+```
+https://www.classtime.com/service/public/Account/login
+https://www.classtime.com/service/public/Account/getPublicUserProfile
+```
+
+## Role permissions
+
+Each service and service's methods contain an annotation limiting access to specific roles. And of course, methods annotations have higher priority. For example, in the [interface sample](https://github.com/gopollock/public-api/tree/main#interface-sample), there are:
+- `AccountService` with `@Public` service annotation, which makes all methods available for all.
+- `login` method without an annotation. And as a result, it uses `AccountService` scope - `@Public`.
+- `getPublicUserProfile` method with `@RequireRole(Role.Teacher)`. Only teachers are allowed to call this endpoint.
+
+### Auth Header
+
+To get access scope of some roles, you need to have an access token tied to an account with that role.  
+Your access token must be sent in Auth header with a request in the following schema:
+
+```
+Authorization: JWT ${accessToken}
+```
+
+#### Get testing access token
+The easiest way to get an access token for some test purposes is through classtime website:
+1. Go to a [Classtime website](https://www.classtime.com/auth/login) and authorize using any method (e.g. with Google).
+2. Open "classtime.com" cookies in the developer console of the browser. (see e.g. [how to do it in Google Chrome](https://developer.chrome.com/docs/devtools/application/cookies/#open))
+3. Copy a value of a cookie with the name: "service-jwt-0". It is your's teacher account access token.
+
+## JSON requests
+
+JSON requests are the easiest point to start. However, we are recommending to use binary requests.  
+Note: **POST is the only supported method**
+
+**Public AccountService.login call**:
+```
+curl -v https://www.classtime.com/service/public/Account/login \
+    -X POST \
+    -H 'Content-Type: application/json' \
+    -H 'Accept: application/json,*/*' \
+    -d '{ provider: "Google", role: "Teacher" }'
+```
+```
+Response: { "redirectUrl": "https://accounts.google.com..." }
+```
+
+**Restricted AccountService.getPublicUserProfile call**:
+Pre-requirement: you must know your account id and access token. You could obtain it following [this small guide](https://github.com/gopollock/public-api/blob/main/flow-examples/my-account-id.md).
+
+```
+curl -v https://www.classtime.com/service/public/Account/getPublicUserProfile \
```

Comment:
Some URLs might not work valid now. Reason:
1. We do not support `service/public` yet
2. getPublicUserProfile and getMyAccountInfo renames are not released yet on the prod

So, I will wait some time for prod and this doc to align.